lib/galaxy/tool_util/xsd/galaxy.xsd

```diff
@@ -2739,6 +2751,123 @@ $attribute_list::5
+      <xs:annotation>
+        <xs:documentation xml:lang="en">The absolute tolerance to be used for the ``mean_intensity`` and ``center_of_mass`` assertions (defaults to ``0.01``).</xs:documentation>
+      </xs:annotation>
+    </xs:attribute>
+    <xs:attribute name="channel" type="xs:integer" use="optional">
+      <xs:annotation>
+        <xs:documentation xml:lang="en">Restricts the ``mean_intensity``, ``mean_intensity_min``, ``mean_intensity_max``, and ``center_of_mass`` assertions to a specific channel of the image (where the value ``0`` corresponds to the first image channel).</xs:documentation>
+      </xs:annotation>
+    </xs:attribute>
+  </xs:complexType>
+  <xs:complexType name="AssertImageHasLabels">
+    <xs:annotation>
+      <xs:documentation xml:lang="en"><![CDATA[
+Asserts the labels in a label image comply with the specified constraints.
+
+```xml
+<number_of_objects number_of_objects="187" mean_object_size="111.87" eps="0.01" exclude_labels="0" />
+```
+
+$attribute_list::5
+]]>
+      </xs:documentation>
+    </xs:annotation>
+    <xs:attribute name="number_of_objects" type="xs:integer" use="optional">
+      <xs:annotation>
+        <xs:documentation xml:lang="en">The required number of unique labels in the image. It is assumed that each individual object corresponds to a unique label.</xs:documentation>
+      </xs:annotation>
+    </xs:attribute>
+    <xs:attribute name="mean_object_size" type="xs:float" use="optional">
+      <xs:annotation>
+        <xs:documentation xml:lang="en">The required mean size of the objects in the image, where the size of an object is measured by the number of pixels. It is assumed that each individual object corresponds to a unique label.</xs:documentation>
+      </xs:annotation>
+    </xs:attribute>
+    <xs:attribute name="mean_object_size_min" type="xs:float" use="optional">
+      <xs:annotation>
+        <xs:documentation xml:lang="en">A lower bound of the required mean size of the objects in the image, where the size of an object is measured by the number of pixels. It is assumed that each individual object corresponds to a unique label.</xs:documentation>
+      </xs:annotation>
+    </xs:attribute>
+    <xs:attribute name="mean_object_size_max" type="xs:float" use="optional">
+      <xs:annotation>
+        <xs:documentation xml:lang="en">An upper bound of the required mean size of the objects in the image, where the size of an object is measured by the number of pixels. It is assumed that each individual object corresponds to a unique label.</xs:documentation>
+      </xs:annotation>
+    </xs:attribute>
+    <xs:attribute name="eps" type="xs:float" use="optional" default="0.01">
+      <xs:annotation>
+        <xs:documentation xml:lang="en">The absolute tolerance to be used for the ``mean_object_size`` assertion (defaults to 0.01).</xs:documentation>
+      </xs:annotation>
+    </xs:attribute>
+    <xs:attribute name="exclude_labels" type="xs:string" use="optional">
```

Comment:
Could it be useful to have `include_labels` as well?

Reply:
Not sure what the use case for this would be?

Reply:
Ah, you mean `include_labels` as in *only* include those labels. Gotcha.

Reply:
Done this in 00485ccc5294815c06e1d4d51d92d487f4133c65. The attribute is named simply `labels`, I think `include_labels` could be confusing (at least it confused me, like *"what's the point of including a label that is not present in the image?"*).